spec.bs

```diff
@@ -1522,6 +1533,23 @@ To <dfn>convert a modifier to a string</dfn> given a [=part/modifier=] |modifier
   1. Return |dummyURL|'s [=url/host=].
 </div>
 
+<div algorithm>
+  To <dfn>canonicalize an IPv6 hostname</dfn> given a string |value|:
+
+  1. Let |result| be the empty string.
+  1. [=list/For each=] |code point| in |value| interpreted as a sequence of [=/code points=]:
+    1. If all of the following are true:
+      <ul>
+        <li>|code point| is not an [=ASCII hex digit=];</li>
+        <li>|code point| is not U+005B (`[`);</li>
+        <li>|code point| is not U+005D (`]`); and</li>
+        <li>|code point| is not U+003A (`:`),</li>
+      </ul>
+      <p>then throw a {{TypeError}}.
+    1. Append the result of running [=ASCII lowercase=] given "|code point|" to the end of |result|.
```

Comment:
Hmm. This "string literal with a variable" syntax is a bit weird; I thought you were using scare quotes.

A few ideas:

- Just use ASCII lowercase with an implicit conversion from code point to string. Infra can eventually define such an implicit conversion.
- Be very explicit, e.g. "Let _stringForCodePoint_ be a string consisting of the single code point _code point_" and then do that.
- Inline the definition of ASCII lowercase, i.e. "If _code point_ is an ASCII upper alpha, then append the corresponding code point in ASCII lower alpha to _result_. Otherwise, append _code point_ to _result_.

Probably the first is nicest.